plugin.yaml

```diff
@@ -10,62 +10,281 @@ plugins:
+    derived_from: cloudify.nodes.VirtualIP
+    properties:
+      use_external_resource:
+        description: >
+          Indicate whether the resource exists or if Cloudify should create the resource.
+        type: boolean
+        default: false
+      gcp_config:
+        description: >
+          A dictionary of values to pass to authenticate with the GCP API.
+        default: {}
+
 relationships:
-  cloudify.gcp.relationships.connected_to:
+  cloudify.gcp.relationships.instance_connected_to_security_group:
     derived_from: cloudify.relationships.connected_to
+    source_interfaces:
+      cloudify.interfaces.relationship_lifecycle:
+        establish:
+          implementation: gcp_plugin.plugin.tasks.add_instance_tag
+          inputs:
+            gcp_config:
+              default: { get_property: [SOURCE, gcp_config] }
+            instance_name: 
+              default: { get_attribute: [SOURCE, gcp_name] }
+            tag:
+              default: { get_attribute: [TARGET, targetTags] }
+        unlink:
+          implementation: gcp_plugin.plugin.tasks.remove_instance_tag
+          inputs:
+            gcp_config:
+              default: { get_property: [SOURCE, gcp_config] }
+            instance_name: 
+              default: { get_attribute: [SOURCE, gcp_name] }
+            tag:
+              default: { get_attribute: [TARGET, targetTags] }
 
-  cloudify.gcp.relationships.contained_in:
-    derived_from: cloudify.relationships.contained_in
+  cloudify.gcp.relationships.instance_connected_to_ip:
+    derived_from: cloudify.relationships.connected_to
+    source_interfaces:
+      cloudify.interfaces.relationship_lifecycle:
+        establish:
+          implementation: gcp_plugin.plugin.tasks.add_external_ip
+          inputs:
+            gcp_config:
+              default: { get_property: [SOURCE, gcp_config] }
+            instance_name: 
+              default: { get_attribute: [SOURCE, gcp_name] }
+        unlink:
+          implementation: gcp_plugin.plugin.tasks.remove_external_ip
+          inputs:
+            gcp_config:
+              default: { get_property: [SOURCE, gcp_config] }
+            instance_name: 
+              default: { get_attribute: [SOURCE, gcp_name] }
 
+  cloudify.gcp.relationships.instance_connected_to_keypair:
+    derived_from: cloudify.relationships.connected_to
+    source_interfaces:
+      cloudify.interfaces.relationship_lifecycle:
+        establish:
+          implementation: gcp_plugin.plugin.tasks.save_private_key
+          inputs:
+            gcp_config:
+              default: { get_property: [SOURCE, gcp_config] }
+            private_key: 
+              default: { get_attribute: [TARGET, gcp_private_key] }
+            user: 
+              default: { get_property: [TARGET, user] }
+            private_key_path: 
+              default: { get_property: [TARGET, private_key_path] }
+        unlink:
+          implementation: gcp_plugin.plugin.tasks.delete_private_key
```

Comment:
same applies here (if you take into account my comment on tasks.py to call delete_private_key at the end of delete_keypair)